IpmiFeaturePkg/Samples/IpmiStatusReporter/IpmiStatusReporter.c

```diff
@@ -0,0 +1,62 @@
+/** @file
+  Sample common code for sending EfiReportStatus to IPMI SEL.
+
+  Copyright (c) Microsoft Corporation
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <IpmiStatusReporter.h>
+
+/**
+  Sample implementation for logging a Status Code event to the System Event Log (SEL).
+
+  @param[in]  CodeType   The status code type.
+  @param[in]  Value      The status code value.
+  @param[in]  Instance   The status code instance number.
+  @param[in]  CallerId   Pointer to a GUID that identifies the caller of this
+                         function. This is an optional parameter that may be NULL.
+  @param[in]  Data       Pointer to the extended data buffer. This is an optional
+                         parameter that may be NULL.
+
+  @retval     EFI Status Code.
+**/
+EFI_STATUS
+EFIAPI
+IpmiReportStatusCode (
+  IN EFI_STATUS_CODE_TYPE   CodeType,
+  IN EFI_STATUS_CODE_VALUE  Value,
+  IN UINT32                 Instance,
+  IN EFI_GUID               *CallerId OPTIONAL,
+  IN EFI_STATUS_CODE_DATA   *Data OPTIONAL
+  )
+
+{
+  EFI_STATUS            Status;
+  UINT16                RecordId;
+  SEL_STATUS_CODE_DATA  EntryData;
+
+  //
+  // TODO: Platform specific logic for deciding what kind of events to be logged.
```

Comment:
nit: Would it be better to implement this filter and publish this as a common driver? We can abstract a filter interface if we would like to provide such flexibility.

Reply:
I'll add a comment to this effect